### PR TITLE
Issue #7 by kumikoono, naoi: Assertions for no Error Messages

### LINKED ISSUE #7: Assertions for no Error Messages
The existing assertions are checking if a certain text is visible or not-visible in the success/warning/error message. However, there are no assertions to see if no success/warning/error messages are visible.

### PR DESCRIPTION
**Pull request by kumikoono**
To assert no error/warning messages visible

### CODE REVIEW

**Comment by onotmil** [comment]
It looks good to me.

File: src/Drupal/DrupalExtension/Context/MessageContext.php [hunk 1]
```diff
@@ -361,4 +361,55 @@ private function assertNot($message, $selectorId, $exceptionMsg)
...
+    if ($selector == 'success') {
+      $selector_id = 'success_message_selector';
+    }
+    else if (selector == 'error') {
+      $selector_id = 'error_message_selector';
+    }
+    else {
+      $selector_id = 'warning_message_selector';
+    }
+    $this->assertNotVisible($selector_id);
```
Comment (naoi) [inline src/Drupal/DrupalExtension/Context/MessageContext.php]: @kumikoono This should be:

```suggestion
  public function assertSelectorNotVisible($selector): void {

    $selector_id = 'warning_message_selector';

    if ($selector == 'success') {
      $selector_id = 'success_message_selector';
    }

    if (selector == 'error') {
      $selector_id = 'error_message_selector';
    }

    $this->assertNotVisible($selector_id);
```

File: src/Drupal/DrupalExtension/Context/MessageContext.php [hunk 2]
```diff
@@ -361,4 +361,55 @@ private function assertNot($message, $selectorId, $exceptionMsg)
...
+  }
+
+  /**
+   * Checks if the current page contains neither error nor warning messages.
+   *
+   * @Then I should see neither error nor warning messages
+   */
+  public function assertErrorWarningNotVisible():void {
```
Comment (naoi) [inline src/Drupal/DrupalExtension/Context/MessageContext.php]: @kumikoono This should be:

```suggestion
  public function assertErrorWarningNotVisible(): void {
```

File: src/Drupal/DrupalExtension/Context/MessageContext.php [hunk 3]
```diff
@@ -361,4 +361,55 @@ private function assertNot($message, $selectorId, $exceptionMsg)
...
+  /**
+   * Assert the selector is not visible.
+   *
+   * @param string $selector_id
+   *  The selector id to be checked.
+   *
+   * @throws \Behat\Mink\Exception\ExpectationException
+   *   Thrown when the unexpected message is present in the page.
+   */
+  private function assertNotVisible(string $selector_id):void {
```
Comment (naoi) [inline src/Drupal/DrupalExtension/Context/MessageContext.php]: @kumikoono This should be:

```suggestion
  private function assertNotVisible(string $selector_id): void {
```

File: src/Drupal/DrupalExtension/Context/MessageContext.php [hunk 4]
```diff
@@ -361,4 +361,55 @@ private function assertNot($message, $selectorId, $exceptionMsg)
...
+    $selector = $this->getDrupalSelector($selector_id);
+    $exception_msg = "The page '%s' contains the $selector_id '%s': %s";
+    $selector_objects = $this->getSession()->getPage()->findAll("css", $selector);
+    $visible_messages = [];
+    if (!empty($selector_objects)) {
+      foreach ($selector_objects as $selector_object) {
+        $visible_messages += $selector_object->getText();
+      }
+      throw new ExpectationException(sprintf($exception_msg, $this->getSession()->getCurrentUrl(), $selector_id, $visible_messages), $this->getSession()->getDriver());
+    }
```
Comment (naoi) [inline src/Drupal/DrupalExtension/Context/MessageContext.php]: @kumikoono This should be:

```suggestion
    if (empty($selector_objects)) {
      return;
    }

    $visible_messages = [];
    foreach ($selector_objects as $selector_object) {
      if (empty($selector_object)) {
        continue;
      }
      $visible_messages += $selector_object->getText();
    }
    throw new ExpectationException(sprintf($exception_msg, $this->getSession()->getCurrentUrl(), $selector_id, $visible_messages), $this->getSession()->getDriver());
```

**Review by naoi — CHANGES_REQUESTED** [review]

File: src/Drupal/DrupalExtension/Context/MessageContext.php (hunk 3 above)
Comment (kumikoono) [reply src/Drupal/DrupalExtension/Context/MessageContext.php]: Done

File: src/Drupal/DrupalExtension/Context/MessageContext.php (hunk 2 above)
Comment (kumikoono) [reply src/Drupal/DrupalExtension/Context/MessageContext.php]: Done

File: src/Drupal/DrupalExtension/Context/MessageContext.php (hunk 1 above)
Comment (kumikoono) [reply src/Drupal/DrupalExtension/Context/MessageContext.php]: Changed to use switch.

File: src/Drupal/DrupalExtension/Context/MessageContext.php (hunk 4 above)
Comment (kumikoono) [reply src/Drupal/DrupalExtension/Context/MessageContext.php]: Done.

**Comment by kumikoono** [comment]
Thanks for your review. Updated.

**Pull request merged**